pkg/permissions/permissions.go

```diff
@@ -276,6 +281,42 @@ func GetForShareCode(db prefixer.Prefixer, tokenCode string) (*Permission, error
 	return perm, nil
 }
 
+// GetTokenFromShortcode retrieves the token doc for a given sharing shortcode
+func GetTokenFromShortcode(db prefixer.Prefixer, shortcode string) (string, error) {
+	var res couchdb.ViewResponse
+
+	err := couchdb.ExecView(db, consts.PermissionsShareByShortcodeView, &couchdb.ViewRequest{
+		Key:         shortcode,
+		IncludeDocs: true,
+	}, &res)
+	if err != nil {
+		return "", err
+	}
+
+	if len(res.Rows) == 0 {
+		return "", fmt.Errorf("no permission doc for shortcode %v", shortcode)
+	}
+
+	if len(res.Rows) > 1 {
+		return "", fmt.Errorf("Bad state: several permission docs for shortcode %v", shortcode)
+	}
+
+	perm := &Permission{}
+	err = json.Unmarshal(res.Rows[0].Doc, &perm)
```

Comment:
Are you sure about `&perm`? `perm` is already a pointer, it looks weird to use `&` with a pointer. Oh, and you should add a unit test for this function, I don't want to guess if it works!